dynamodb/src/main/scala/zio/dynamodb/ProjectionExpression.scala

```diff
@@ -65,15 +74,22 @@ sealed trait ProjectionExpression { self =>
   def isNull: ConditionExpression      = isType(AttributeValueType.Null)
   def isStringSet: ConditionExpression = isType(AttributeValueType.StringSet)
 
-  private def isType(attributeType: AttributeValueType): ConditionExpression =
+  private def isType(attributeType: AttributeValueType): ConditionExpression = // TODO: private so move down
     ConditionExpression.AttributeType(self, attributeType)
 
   // ConditionExpression with AttributeValue's
 
-  def contains[A](av: A)(implicit t: ToAttributeValue[A]): ConditionExpression                   =
+  // constraint: String OR Set
+  def contains[A](av: A)(implicit t: ToAttributeValue[A]): ConditionExpression =
     ConditionExpression.Contains(self, t.toAttributeValue(av))
-  def beginsWith[A](av: A)(implicit t: ToAttributeValue[A]): ConditionExpression                 =
-    ConditionExpression.BeginsWith(self, t.toAttributeValue(av))
+
+  // constraint: String
+  def beginsWith(av: String)(implicit ev: RefersToString[To]): ConditionExpression = {
+    val _ = ev
```

Comment:
Is this just an ignored value being set? Do we need this here?

Reply:
this is a compile time proof to force type safety, we don't actually use `ev`

Reply:
Ah okay that's really cool! I don't I've seen one of these before.